h2/src/main/org/h2/api/H2Type.java

```diff
@@ -0,0 +1,273 @@
+
+    // Exact numeric data types
+
+    /**
+     * The TINYINT data type.
+     */
+    TINYINT("TINYINT", Value.TINYINT),
+
+    /**
+     * The SMALLINT data type.
+     */
+    SMALLINT("SMALLINT", Value.SMALLINT),
+
+    /**
+     * The INTEGER data type.
+     */
+    INTEGER("INTEGER", Value.INTEGER),
+
+    /**
+     * The BIGINT data type.
+     */
+    BIGINT("BIGINT", Value.BIGINT),
+
+    /**
+     * The NUMERIC data type.
+     */
+    NUMERIC("NUMERIC", Value.NUMERIC),
+
+    // Approximate numeric data types
+
+    /**
+     * The REAL data type.
+     */
+    REAL("REAL", Value.REAL),
+
+    /**
+     * The DOUBLE data type.
+     */
+    DOUBLE("DOUBLE", Value.DOUBLE),
+
+    // Character strings
+
+    /**
+     * The CHAR data type.
+     */
+    CHAR("CHAR", Value.CHAR),
+
+    /**
+     * The VARCHAR data type.
+     */
+    VARCHAR("VARCHAR", Value.VARCHAR),
+
+    /**
+     * The VARCHAR_IGNORECASE data type.
+     */
+    VARCHAR_IGNORECASE("VARCHAR_IGNORECASE", Value.VARCHAR_IGNORECASE),
+
+    /**
+     * The CLOB data type.
+     */
+    CLOB("CLOB", Value.CLOB),
+
+    // Date-time data types
+
+    /**
+     * The DATE data type.
+     */
+    DATE("DATE", Value.DATE),
+
+    /**
+     * The TIME data type.
+     */
+    TIME("TIME", Value.TIME),
+
+    /**
+     * The TIME WITH TIME ZONE data type.
+     */
+    TIME_WITH_TIME_ZONE("TIME WITH TIME ZONE", Value.TIME_TZ),
+
+    /**
+     * The TIMESTAMP data type.
+     */
+    TIMESTAMP("TIMESTAMP", Value.TIMESTAMP),
+
+    /**
+     * The TIMESTAMP WITH TIME ZONE data type.
+     */
+    TIMESTAMP_WITH_TIME_ZONE("TIMESTAMP WITH TIME ZONE", Value.TIMESTAMP_TZ),
+
+    // Intervals
+
+    /**
+     * The INTERVAL YEAR data type.
+     */
+    INTERVAL_YEAR("INTERVAL YEAR", Value.INTERVAL_YEAR),
+
+    /**
+     * The INTERVAL MONTH data type.
+     */
+    INTERVAL_MONTH("INTERVAL MONTH", Value.INTERVAL_MONTH),
+
+    /**
+     * The INTERVAL DAY data type.
+     */
+    INTERVAL_DAY("INTERVAL DAY", Value.INTERVAL_DAY),
+
+    /**
+     * The INTERVAL HOUR data type.
+     */
+    INTERVAL_HOUR("INTERVAL HOUR", Value.INTERVAL_HOUR),
+
+    /**
+     * The INTERVAL MINUTE data type.
+     */
+    INTERVAL_MINUTE("INTERVAL MINUTE", Value.INTERVAL_MINUTE),
+
+    /**
+     * The INTERVAL SECOND data type.
+     */
+    INTERVAL_SECOND("INTERVAL SECOND", Value.INTERVAL_SECOND),
+
+    /**
+     * The INTERVAL YEAR TO MONTH data type.
+     */
+    INTERVAL_YEAR_TO_MONTH("INTERVAL YEAR TO MONTH", Value.INTERVAL_YEAR_TO_MONTH),
+
+    /**
+     * The INTERVAL DAY TO HOUR data type.
+     */
+    INTERVAL_DAY_TO_HOUR("INTERVAL DAY TO HOUR", Value.INTERVAL_DAY_TO_HOUR),
+
+    /**
+     * The INTERVAL DAY TO MINUTE data type.
+     */
+    INTERVAL_DAY_TO_MINUTE("INTERVAL DAY TO MINUTE", Value.INTERVAL_DAY_TO_MINUTE),
+
+    /**
+     * The INTERVAL DAY TO SECOND data type.
+     */
+    INTERVAL_DAY_TO_SECOND("INTERVAL DAY_TO_SECOND", Value.INTERVAL_DAY_TO_SECOND),
```

Comment:
this string looks wrong?

Reply:
Yes, definitely. It's just a some our own name, but it should have the same style as others.

4e615c59563fc19d20080554d41cd5e94526712e